Expanding collapsed columns

VhugoJc opened the pull request:
Fixes #5879 


<img width="563" alt="Screenshot 2023-07-19 at 12 22 14" src="https://github.com/OpenRefine/OpenRefine/assets/69024201/92d149d6-afce-4b88-9785-395e6cd34550">

VhugoJc commented:
I fixed the Issue mentioned by @wetneb in #5869 (:

Abbe98 commented on main/webapp/modules/core/scripts/views/data-table/data-table-view.js:
```diff
@@ -371,12 +371,20 @@ DataTableView.prototype._renderDataTables = function(table, tableHeader) {
-      });
+      DOM.bind( 
+        $(th)
+        .html("<button class='column-header-menu column-header-menu-expand' bind='expandColumn' ></button>")
```
A title like "Expand this column" would be nice.

Abbe98 commented on main/webapp/modules/core/scripts/views/data-table/data-table-view.js:
```diff
@@ -371,12 +371,20 @@ DataTableView.prototype._renderDataTables = function(table, tableHeader) {
   var createColumnHeader = function(column, index) {
     var th = trHead.appendChild(document.createElement("th"));
-    $(th).addClass("column-header").attr('title', column.name);
+    $(th)
```
Unrelated change?

Abbe98 commented on main/webapp/modules/core/styles/views/data-table-view.css:
```diff
@@ -188,6 +188,10 @@ table.data-table td.column-header, table.data-table th.column-header {
   border: none;
 }
 
+.column-header-menu-expand{
```
Nitpick: missing space.

Abbe98 reviewed (dismissed):
Looks solid to me! ⭐

VhugoJc commented:
I changed the label to "Expand this column" and changed the requested changes. 
<img width="434" alt="Screenshot 2023-07-24 at 15 58 18" src="https://github.com/OpenRefine/OpenRefine/assets/69024201/ef8de399-9131-49c6-a347-2315fe97c247">
I hope this is what you were looking for! Let me know if it's good or if you need anything else. Thanks!

wetneb commented on main/webapp/modules/core/scripts/views/data-table/data-table-view.js:
```diff
@@ -373,10 +373,16 @@ DataTableView.prototype._renderDataTables = function(table, tableHeader) {
-      });
+      DOM.bind( 
+        $(th)
+        .attr('title',"Expand this column")
```
Oops, here is a problem I oversaw: we'd need this text to be localized, with our `$.i18n` utility.

VhugoJc replied:
I have changed the text for `$.i18n` utility and I added the text in all the supported languages. Please, tell me if you need anything else 🚀

wetneb reviewed (commented):
Thank you so much for the swift reaction! It is also kind of you to have filled all translations, but we prefer to avoid relying on machine translation for this and we normally only add the English strings when making changes like this one. Translators will be duly notified that this new string is inserted and will have the opportunity to translate it on their own. Could you remove those new translations and only keep the changes to `translation-en.json`? I am really sorry you went through all this work - I should have explained our practices in the first place.

VhugoJc commented:
Don't worry, I perfectly understand!

wetneb reviewed (approved):
Thanks again!

wetneb commented:
Ah, of course the Cypress tests need to be updated to reflect the new representation of collapsed columns. This test suite is documented here: https://openrefine.org/docs/technical-reference/functional-tests
Let me know if you need help with that.

wetneb commented:
@VhugoJc are you still on this? It's nearly good to go, we just need to update the Cypress tests to make sure they still pass after your change.

VhugoJc commented:
ok, let me try again. sorry for the delay.

wetneb commented:
I have changed the tooltip to be "Expand column 'foo'" instead of "Expand this column", so that users are able to tell which column they are about to expand. This also made it easier to adapt the Cypress tests, since they were relying on this tooltip to click the column headers to expand the column.

Updating the Cypress tests is definitely some work, but it's nice that doing that encourages us to improve the UX! And it's not the first time it happens to me.
